dm/syncer/checkpoint.go

```diff
@@ -100,9 +100,11 @@ func (b *binlogPoint) save(location binlog.Location, ti *model.TableInfo) error
 	b.Lock()
 	defer b.Unlock()
 
-	if binlog.CompareLocation(location, b.savedPoint.location, b.enableGTID) < 0 {
-		// support to save equal location, but not older location
-		return terror.ErrCheckpointSaveInvalidPos.Generate(location, b.savedPoint.location)
+	if ce := log.L().Check(zap.DebugLevel, "check binlog point regression"); ce != nil {
```

Comment:
please add some comment to explain why code other than log is guarded by log level.